cpp/include/cudf/io/types.hpp

```diff
@@ -201,20 +201,33 @@ enum dictionary_policy {
 };
 
 /**
- * @brief Detailed name information for output columns.
+ * @brief Detailed name (and optionally nullability) information for output columns.
  *
  * The hierarchy of children matches the hierarchy of children in the output
  * cudf columns.
  */
 struct column_name_info {
   std::string name;                        ///< Column name
+  std::optional<bool> is_nullable;         ///< Column nullability
   std::vector<column_name_info> children;  ///< Child column names
   /**
    * @brief Construct a column name info with a name and no children
    *
    * @param _name Column name
    */
   column_name_info(std::string const& _name) : name(_name) {}
+
+  /**
+   * @brief Construct a column name info with a name, nullabilty, and no children
+   *
+   * @param _name Column name
+   * @param _is_nullable True if column is nullable
+   */
+  column_name_info(std::string const& _name, bool _is_nullable)
```

Comment:
Default to unspecified nullability?
```suggestion
  column_name_info(std::string const& _name, std::optional<bool> _is_nullable = std::nullopt)
```

Reply:
I think that will collide with the other ctor?

Reply:
made me look!